galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos.manager/src/main/java/dev/galasa/zos/internal/ZosPoolPorts.java

```diff
@@ -0,0 +1,68 @@
+package dev.galasa.zos.internal;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.framework.spi.DssDeletePrefix;
+import dev.galasa.framework.spi.DssUpdate;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+import dev.galasa.framework.spi.DynamicStatusStoreMatchException;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IResourcePoolingService;
+import dev.galasa.framework.spi.InsufficientResourcesAvailableException;
+import dev.galasa.zos.ZosManagerException;
+import dev.galasa.zos.internal.properties.PoolPorts;
+
+public class ZosPoolPorts {
+	
+    private static final Log logger = LogFactory.getLog(ZosPoolPorts.class);
+
+	private final ZosManagerImpl 			 manager;
+	private final IDynamicStatusStoreService dss;
+	private final IResourcePoolingService 	 rps;	
+		
+	public ZosPoolPorts(ZosManagerImpl manager, IDynamicStatusStoreService dss, IResourcePoolingService rps) throws ZosManagerException {
+		this.manager = manager;
+		this.dss = dss;
+		this.rps = rps;
+	}
+	
+	public String allocatePort(String image) throws ZosManagerException {
+
+		// The the pool of ports for the specific image
+		List<String> resourceStrings = PoolPorts.get(image);
+		
+		String thePort = null;
+		try {
+			List<String> zosPorts = this.rps.obtainResources(resourceStrings, null, 1, 1, dss, "zosport");
```

Comment:
if this is empty after the call to obtain resources then thePort will be returned as null